packages/io-messages-common/src/adapters/lollipop/__tests__/lollipop-client.test.ts

```diff
@@ -0,0 +1,104 @@
+import { describe, expect, it, vi } from "vitest";
+
+import {
+  aProblemJson,
+  aSignatureInput,
+  anAssertionRef,
+  anLcParams,
+} from "../__mocks__/lollipop.js";
+import LollipopClient from "../lollipop-client.js";
+
+const apiKey = "anApiKey";
+const baseUrl = "https://mockurl.com";
+const client = new LollipopClient(apiKey, baseUrl);
+
+describe("LollipopClient", () => {
+  it("returns a valid LcParams on successful request", async () => {
+    const fetchSpy = vi.spyOn(globalThis, "fetch").mockResolvedValueOnce({
+      json: async () => anLcParams,
+      ok: true,
+      status: 200,
+    } as Response);
+
+    const lcParams = await client.generateLCParams(
+      anAssertionRef,
+      aSignatureInput,
+    );
+
+    expect(lcParams).toEqual(anLcParams);
+    expect(fetchSpy).toHaveBeenCalledWith(
+      `${baseUrl}/pubKeys/${anAssertionRef}/generate`,
+      expect.objectContaining({
+        body: JSON.stringify({ operationId: aSignatureInput }),
+        headers: expect.objectContaining({
+          "X-Functions-Key": apiKey,
+          "content-type": "application/json",
+        }),
+        method: "POST",
+      }),
+    );
+  });
+
+  it("throws an error when the API responds with an error", async () => {
```

Comment:
In those 2 `it` we are not sure that the `expect` in the `catch` block will be executed, if for some reason the generateLCParams does not throw then the expect would be ignored.
I would suggest to use a different approach as we did [here](https://github.com/pagopa/io-messages/blob/main/apps/services-func/src/functions/OnFailedProcessMessage/__tests__/handler.test.ts#L129-L129).

This way we ensure that the generateLCParams failed with the expected error